File: day 1/ex1.exs
calories_on_elf = elem(File.read("ex1.txt"),1)
|> String.split("\n\n")
|> Enum.map(&(String.split(&1, "\n")))
|> Enum.map(&(List.foldl(&1, 0, fn x,acc ->
  case Integer.parse(x) do
    {parsed, _} -> acc + parsed
    :error -> acc
  end
end )))
|> Enum.sort(:desc)
|> Enum.take(3)
IO.puts(Enum.max(calories_on_elf))
IO.puts(Enum.sum(calories_on_elf))

File: day 1/ex1.rb
calories_on_elf = [ 0 ]
File.foreach("ex1.txt") { |line| 
    line.chomp!
    line.empty? ? calories_on_elf.push(0) : calories_on_elf[-1] += line.to_i;
}
puts calories_on_elf.max;
puts calories_on_elf.sort_by{ |i| -i }.first(3).inject(:+);